crates/rattler_solve/src/libsolv/input.rs

```diff
@@ -0,0 +1,258 @@
+//! Contains business logic that loads information into libsolv in order to solve a conda
+//! environment
+
+use crate::libsolv::util;
+use crate::libsolv::util::PackageExtension;
+use crate::libsolv::wrapper::keys::*;
+use crate::libsolv::wrapper::pool::Pool;
+use crate::libsolv::wrapper::repo::Repo;
+use crate::libsolv::wrapper::repodata::Repodata;
+use crate::libsolv::wrapper::solvable::SolvableId;
+use rattler_conda_types::{GenericVirtualPackage, RepoDataRecord};
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::ffi::{CString, NulError};
+
+/// Adds [`RepoDataRecord`] to this instance
+pub fn add_repodata_records(
+    pool: &Pool,
+    repo: &Repo,
+    repo_datas: &[RepoDataRecord],
+) -> Result<(), NulError> {
+    let data = repo.add_repodata();
+
+    // Get all the IDs
+    let solvable_buildflavor_id = pool.find_interned_str(SOLVABLE_BUILDFLAVOR).unwrap();
```

Comment:
Technically, this can never fail, because these strings are internal to libsolv and always present. If there is something wrong there, a panic seems like the only sensible option.

Reply:
Could you add that in a comment or in an `expect`?